### PR TITLE
Make Integer#<=> spec compliant

### PR DESCRIPTION
**Pull request by richardboehme**
I do not like the way I have to use the same if-condition twice in IntegerObject::cmp. Does somebody have an idea how this code could be done in a nicer way? 🤔 

[#201]

### CODE REVIEW

File: include/natalie.hpp [hunk 1]
```diff
@@ -106,7 +106,7 @@ ArrayObject *block_args_to_array(Env *env, size_t signature_size, size_t argc, V
 
 void arg_spread(Env *env, size_t argc, Value *args, const char *arrangement, ...);
 
-std::pair<Value, Value> coerce(Env *, Value, Value);
+std::pair<Value, Value> coerce(Env *, Value, Value, bool raise = true);
```
Comment (seven1m) [inline include/natalie.hpp:109]: Can we make this an enum like we did with [ConstLookupFailureMode](https://github.com/seven1m/natalie/blob/master/include/natalie/object.hpp#L43-L46)?

File: src/integer_object.cpp [hunk 2]
```diff
@@ -226,11 +226,13 @@ Value IntegerObject::pow(Env *env, Value arg) const {
...
-        if (!arg->is_integer() && !arg->is_float())
-            return NilObject::the();
-    }
+    // Check if we might want to coerce the value
+    if (!arg.is_fast_integer() && !arg->is_integer() && !arg->is_float())
+        arg = Natalie::coerce(env, arg, this, false).second;
+
+    // Check if compareable
+    if (!arg.is_fast_integer() && !arg->is_integer() && !arg->is_float())
+        return NilObject::the();
```
Comment (seven1m) [inline src/integer_object.cpp:235]: Sadly I cannot think of a technique to eliminate this code duplication that itself would not make the code harder to read. 😞 I considered a lambda, but I don't think the added complexity is worth eliminating one duplicated line. 🤷

File: include/natalie.hpp (hunk 1 above)
Comment (richardboehme) [reply include/natalie.hpp:109]: Yeah of course! That's a great idea 😄

File: src/integer_object.cpp (hunk 2 above)
Comment (richardboehme) [reply src/integer_object.cpp:235]: Yeah I thought about a lambda as well... But what do you think about a short method? Something like `is_comparable_with`?

Comment (TheGrizzlyDev) [reply src/integer_object.cpp:235]: I think a lambda would bring less overhead and the compiler is very likely to inline it, so I don't think it'd be too bad :)

Comment (richardboehme) [reply src/integer_object.cpp:239]: I added a lambda for it! @seven1m If you think it's too complex you can just remove the last commit. I'll be away for the next few days so I just did that in advance 😄 🎄

File: include/natalie.hpp (hunk 1 above)
Comment (richardboehme) [reply include/natalie.hpp]: I added it! The name I chose is a bit mouthful to me but I couldn't think of a better one...

**Pull request merged**